### ISSUE TITLE
[I] Periods and commas 'slice' the space between words or sentences

### ISSUE THREAD

**Issue by gabrivera** [pbek/QOwnNotes#2117]
#### Expected behaviour

Typing "This is a sentence. This is another sentence." and seeing a space between the period  "." and the "T" from "This is another sentence".

#### Actual behaviour

The sentences are sort of spliced together such that it reads "This is a sentence.This is another sentence." Notice there is no space between the period "." and the "T" from "This is another sentence."

#### Steps to reproduce

Using MacOS 11.2.3 , install QOwnNotes from Homebrew. Launch and, in a new note, type down a period or a comma. Then type a space. Then type any letter.

#### Relevant log output in the Log panel

<details><summary>Expand</summary>

```
[20:35:26] [status] Se guardó 1 nota en disco
[20:35:27] [status] La nota fue modificada externamente: /Users/gabrielriverabarreto/Documents/Joplinz/Md export 2021-4-10 22-41/Life/Self-Improvement & Self-Acceptance/eBlug 4. JopzZettz.md
```
</details>

#### Information about the application, settings and environment

<details><summary>Expand</summary>

QOwnNotes Debug Information
===========================

## General Info

**Current Date**: `lun abr. 26 11:38:39 2021`
**Version**: `21.4.4`
**Build date**: `Apr 22 2021`
**Build number**: `830`
**Platform**: `macosx`
**Operating System**: `macOS 11.2`
**Build architecture**: `x86_64`
**Current architecture**: `x86_64`
**Release**: `GitHub Actions`
**Qt Version (build)**: `5.15.2`
**Qt Version (runtime)**: `5.15.2`
**Portable mode**: `no`
**Settings path / key**: `/Users/gabrielriverabarreto/Library/Preferences/com.pbe.QOwnNotes.plist`
**Application database path**: `/Users/gabrielriverabarreto/Library/Application Support/PBE/QOwnNotes/QOwnNotes.sqlite`
**Application arguments**: `/Applications/QOwnNotes.app/Contents/MacOS/QOwnNotes`
**Qt Debug**: `0`
**Locale (system)**: `es_GT`
**Locale (interface)**: *empty*
**Primary screen resolution**: `1440x900`
**Icon theme**: `breeze-qownnotes`
**Notes in current note folder**: `100`
**Calendar items**: `0`
**Enabled scripts**: `0`

## Server Info

**serverUrl**: *empty*
**appIsValid**: `no`
**notesPathExists**: *empty*
**connectionErrorMessage**: *empty*

## Spellchecking

**Enabled**: `false`
**Selected language**: *empty*
**Language codes**: *empty*
**Language names**: *empty*
**Application dictionaries path**: `/Users/gabrielriverabarreto/Library/Application Support/PBE/QOwnNotes/dicts`

## Note folders

**currentNoteFolderId**: `2`

### Note folder `predeterminado`

**id**: `1`
**isCurrent**: `no`
**activeTagId**: `0`
**localPath**: `/Users/gabrielriverabarreto/Documents/Joplinz/Md export 2021-4-10 22-41`
**remotePath**: `Notes`
**cloudConnectionId**: `1`
**isShowSubfolders**: `yes`
**isUseGit**: `no`
**activeNoteSubFolder name**: `Políticas Públicas 2`
**database file**: `/Users/gabrielriverabarreto/Documents/Joplinz/Md export 2021-4-10 22-41/notes.sqlite`

### Note folder `predeterminado`

**id**: `2`
**isCurrent**: `yes`
**activeTagId**: `0`
**localPath**: `/Users/gabrielriverabarreto/Documents/Joplinz/Md export 2021-4-10 22-41`
**remotePath**: `Notes`
**cloudConnectionId**: `1`
**isShowSubfolders**: `yes`
**isUseGit**: `no`
**activeNoteSubFolder name**: *empty*
**database file**: `/Users/gabrielriverabarreto/Documents/Joplinz/Md export 2021-4-10 22-41/notes.sqlite`

### Cloud connection ``

**id**: `1`
**isCurrent**: `yes`
**serverUrl**: *empty*
**username**: *empty*

## Settings

**AKLastCheckInAttemptDate** (QDateTime): `2019-11-01T16:25:28.995`
**AKLastCheckInSuccessDate** (QDateTime): `2019-11-01T16:25:30.071`
**AKLastEmailListRequestDateKey** (QDateTime): `2021-04-25T22:47:40.616`
**AKLastIDMSEnvironment** (qlonglong): `0`
**ActiveNoteHistoryItem** (NoteHistoryItem): `<binary data>`
**Beats Electronics Beats Studio³** (QString): `1`
**Country** (QString): `GT`
**Debug/fakeOldVersionNumber** (bool): `false`
**Debug/fileLogging** (bool): `false`
**DistractionFreeMode/isEnabled** (bool): `true`
**DistractionFreeMode/menuBarGeometry** (QByteArray): `<binary data>`
**DistractionFreeMode/menuBarHeight** (qlonglong): `0`
**DistractionFreeMode/menuBarVisible** (bool): `true`
**DistractionFreeMode/windowState** (QByteArray): `<binary data>`
**Editor/CurrentSchemaKey** (QString): `EditorColorSchema-6033d61b-cb96-46d5-a3a8-20d5172017eb`
**Editor/autoBracketClosing** (bool): `true`
**Editor/autoBracketRemoval** (bool): `true`
**Editor/centerCursor** (bool): `true`
**Editor/disableCursorBlinking** (bool): `false`
**Editor/editorWidthInDFMOnly** (bool): `true`
**Editor/highlightCurrentLine** (bool): `true`
**Editor/indentSize** (qlonglong): `4`
**Editor/removeTrailingSpaces** (bool): `false`
**Editor/useTabIndent** (bool): `false`
**Editor/vimMode** (bool): `false`
**FileDialog/LastPath** (QString): `/Users/gabrielriverabarreto/Desktop`
**FileDialog/LastPath-NoteHTMLExport** (QString): `/Users/gabrielriverabarreto/Desktop`
**HP ENVY Photo 7800 series** (QString): `1`
**IssueAssistantDialog/geometry** (QByteArray): `<binary data>`
**LastUpdateCheck** (QDateTime): `2021-04-26T10:57:47.661`
**LogWidget/criticalLog** (bool): `true`
**LogWidget/debugLog** (bool): `true`
**LogWidget/fatalLog** (bool): `true`
**LogWidget/infoLog** (bool): `true`
**LogWidget/scriptingLog** (bool): `true`
**LogWidget/statusLog** (bool): `true`
**LogWidget/warningLog** (bool): `true`
**Logitech USB Receiver** (QString): `1`
**MainWindow/geometry** (QByteArray): `<binary data>`
**MainWindow/mainToolBar.iconSize** (qlonglong): `32`
**MainWindow/menuBarGeometry** (QByteArray): `<binary data>`
**MainWindow/noteTextEdit.code.font** (QString): `.AppleSystemUIFontMonospaced,10,-1,5,50,0,0,0,0,0`
**MainWindow/noteTextEdit.font** (QString): `.AppleSystemUIFont,13,-1,5,50,0,0,0,0,0`
**MainWindow/noteTextView.code.font** (QString): `.AppleSystemUIFontMonospaced,10,-1,5,50,0,0,0,0,0`
**MainWindow/noteTextView.font** (QString): `.AppleSystemUIFont,13,-1,5,50,0,0,0,0,0`
**MainWindow/noteTextView.ignoreCodeFontSize** (bool): `true`
**MainWindow/noteTextView.rtl** (bool): `false`
**MainWindow/noteTextView.underline** (bool): `true`
**MainWindow/noteTextView.useEditorStyles** (bool): `true`
**MainWindow/noteTextView.useInternalExportStyling** (bool): `true`
**MainWindow/showNotesFromAllNoteSubFolders** (bool): `true`
**NoteDiffDialog/geometry** (QByteArray): `<binary data>`
**NoteFolder-1/NoteTabNameList** (QStringList): `Clase 7`
**NoteFolder-1/NoteTabStickinessList** (QStringList): *empty*
**NoteFolder-1/NoteTabSubFolderPathDataList** (QStringList): 
```
Maestría
Políticas Públicas 2
```
**NoteFolder-2/NoteTabNameList** (QStringList): `eBlug 4. JopzZettzqOwn`
**NoteFolder-2/NoteTabStickinessList** (QStringList): *empty*
**NoteFolder-2/NoteTabSubFolderPathDataList** (QStringList): 
```
Life
Self-Improvement & Self-Acceptance
```
**NoteHistory-1** (QVariantList): `<variant list with 7 item(s)>`
**NoteHistory-2** (QVariantList): `<variant list with 4 item(s)>`
**NoteHistoryCurrentIndex-1** (qlonglong): `6`
**NoteHistoryCurrentIndex-2** (qlonglong): `3`
**PKSecureElementAvailableFlagsByHost** (qlonglong): `3`
**PiwikClientId** (QString): `<hidden>`
**ScriptRepositoryDialog/mainSplitterState** (QByteArray): `<binary data>`
**SearchEngineId** (qlonglong): `2`
**SettingsDialog/geometry** (QByteArray): `<binary data>`
**SettingsDialog/mainSplitterState** (QByteArray): `<binary data>`
**ShowSystemTray** (bool): `false`
**Silicon-Power PHD 3/0 Silicon-Power** (QString): `1`
**StartHidden** (bool): `false`
**WebAutomaticSpellingCorrectionEnabled** (bool): `true`
**WelcomeDialog/geometry** (QByteArray): `<binary data>`
**Yubico YubiKey OTP+FIDO+CCID** (QString): `1`
**_HIHideMenuBar** (bool): `false`
**acceptAllExternalModifications** (bool): `false`
**allowNoteEditing** (bool): `true`
**allowOnlyOneAppInstance** (bool): `false`
**appMetrics/disableAppHeartbeat** (bool): `false`
**appMetrics/disableTracking** (bool): `false`
**appMetrics/notificationShown** (bool): `true`
**automaticNoteFolderDatabaseClosing** (bool): `false`
**checkSpelling** (bool): `false`
**closeTodoListAfterSave** (bool): `false`
**cryptoKey** (qlonglong): `<hidden>`
**currentNoteFolderId** (qlonglong): `2`
**currentWorkspace** (QString): `f8f2e239-da46-4cd3-9f91-a0920233714a`
**cursorWidth** (qlonglong): `1`
**customNoteFileExtensionList** (QStringList): *empty*
**darkMode** (bool): `false`
**darkModeColors** (bool): `false`
**darkModeIconTheme** (bool): `false`
**darkModeTrayIcon** (bool): `false`
**defaultNoteFileExtension** (QString): `md`
**disableAutomaticUpdateDialog** (bool): `true`
**disableSavedSearchesAutoCompletion** (bool): `false`
**dockWasInitializedOnce** (bool): `true`
**enableNoteTree** (bool): `false`
**enableSocketServer** (bool): `true`
**enableWebAppSupport** (bool): `false`
**externalEditorPath** (QString): *empty*
**fullyHighlightedBlockquotes** (bool): `false`
**gitCommitInterval** (qlonglong): `30`
**gitExecutablePath** (QString): *empty*
**gitLogCommand** (QString): *empty*
**guiFirstRunInit** (bool): `true`
**ignoreAllExternalModifications** (bool): `false`
**ignoreAllExternalNoteFolderChanges** (bool): `false`
**ignoreNoteSubFolders** (QString): `^\.`
**imageScaleDown** (bool): `false`
**imageScaleDownMaximumHeight** (qlonglong): `1024`
**imageScaleDownMaximumWidth** (qlonglong): `1024`
**insertTimeFormat** (QString): *empty*
**interfaceFontSize** (qlonglong): `11`
**interfaceLanguage** (QString): *empty*
**internalIconTheme** (bool): `false`
**itemHeight** (qlonglong): `18`
**legacyLinking** (bool): `false`
**localTrash/autoCleanupDays** (qlonglong): `30`
**localTrash/autoCleanupEnabled** (bool): `true`
**localTrash/supportEnabled** (bool): `true`
**markdownHighlightingEnabled** (bool): `true`
**navigationPanelHideSearch** (bool): `false`
**networking/ignoreSSLErrors** (bool): `true`
**networking/proxyType** (qlonglong): `2`
**newNoteAskHeadline** (bool): `false`
**noteEditIsCentralWidget** (bool): `true`
**noteListPreview** (bool): `false`
**noteSaveIntervalTime** (qlonglong): `10`
**noteSubfoldersPanelDisplayAsFullTree** (bool): `true`
**noteSubfoldersPanelHideSearch** (bool): `false`
**noteSubfoldersPanelOrder** (qlonglong): `0`
**noteSubfoldersPanelShowFullPath** (bool): `false`
**noteSubfoldersPanelShowNotesRecursively** (bool): `false`
**noteSubfoldersPanelShowRootFolderName** (bool): `true`
**noteSubfoldersPanelSort** (qlonglong): `0`
**notesPanelOrder** (qlonglong): `0`
**notesPanelSort** (qlonglong): `1`
**notesPath** (QString): `/Users/gabrielriverabarreto/Documents/Joplinz/Md export 2021-4-10 22-41`
**notifyAllExternalModifications** (bool): `false`
**overrideInterfaceFontSize** (bool): `false`
**ownCloud/supportEnabled** (bool): `false`
**ownCloud/todoCalendarBackend** (qlonglong): `3`
**ownCloud/todoCalendarCalDAVPassword** (QString): `<hidden>`
**ownCloud/todoCalendarCalDAVServerUrl** (QString): *empty*
**ownCloud/todoCalendarCalDAVUsername** (QString): *empty*
**ownCloud/todoCalendarCloudConnectionId** (qlonglong): `1`
**ownCloud/todoCalendarDisplayNameList** (QStringList): *empty*
**ownCloud/todoCalendarEnabledList** (QStringList): *empty*
**ownCloud/todoCalendarEnabledUrlList** (QStringList): *empty*
**ownCloud/todoCalendarUrlList** (QStringList): *empty*
**previousWorkspace** (QString): `b1f1e949-4525-49d0-b0cd-2387aaaf78b1`
**restoreCursorPosition** (bool): `false`
**restoreLastNoteAtStartup** (bool): `true`
**restoreNoteTabs** (bool): `true`
**showMatches** (bool): `true`
**showMenuBar** (bool): `true`
**showStatusBar** (bool): `true`
**systemIconTheme** (bool): `false`
**taggingShowNotesRecursively** (bool): `false`
**tagsPanelHideNoteCount** (bool): `false`
**tagsPanelHideSearch** (bool): `false`
**tagsPanelOrder** (qlonglong): `0`
**tagsPanelSort** (qlonglong): `0`
**todoCalendarSupport** (bool): `true`
**toolbar/1/items** (QStringList): `action_New_note, action_Find_note, action_Remove_note, action_Open_note_in_external_editor, actionShow_local_trash, actionAllow_note_editing, , action_Back_in_note_history, action_Forward_in_note_history, , action_Find_text_in_note, actionReplace_in_current_note, , actionShow_versions, actionShow_trash, actionShare_note, , actionOpen_List`
**toolbar/1/name** (QString): `mainToolBar`
**toolbar/1/title** (QString): `barra de herramientas principal`
**toolbar/2/items** (QStringList): `actionFormat_text_bold, actionFormat_text_italic, actionStrike_out_text, actionInsert_code_block, actionInsert_block_quote`
**toolbar/2/name** (QString): `formattingToolbar`
**toolbar/2/title** (QString): `barra de formato`
**toolbar/3/items** (QStringList): `actionInsert_Link_to_note, actionInsert_image, actionInsert_current_time`
**toolbar/3/name** (QString): `insertingToolbar`
**toolbar/3/title** (QString): `barra de inserción`
**toolbar/4/items** (QStringList): `action_Encrypt_note, actionEdit_encrypted_note, actionDecrypt_note`
**toolbar/4/name** (QString): `encryptionToolbar`
**toolbar/4/title** (QString): `barra de cifrado`
**toolbar/5/items** (QStringList): `actionWorkspaceComboBox, actionStore_as_new_workspace, actionRemove_current_workspace, actionRename_current_workspace, actionSwitch_to_previous_workspace, actionUnlock_panels, , actionToggle_distraction_free_mode, action_Increase_note_text_size, action_Decrease_note_text_size, action_Reset_note_text_size`
**toolbar/5/name** (QString): `windowToolbar`
**toolbar/5/title** (QString): `barra de ventana`
**toolbar/6/items** (QStringList): `action_Quit`
**toolbar/6/name** (QString): `quitToolbar`
**toolbar/6/title** (QString): `quitar barra`
**toolbar/size** (qlonglong): `6`
**useNoteFolderButtons** (bool): `false`
**useUNIXNewline** (bool): `false`
**webAppClientService/serverUrl** (QString): `wss://app.qownnotes.org`
**webAppClientService/token** (QString): `<hidden>`
**webSocketServerService/bookmarksNoteName** (QString): `Bookmarks`
**webSocketServerService/bookmarksTag** (QString): `bookmarks`
**webSocketServerService/port** (qlonglong): `22222`
**workspace-b1f1e949-4525-49d0-b0cd-2387aaaf78b1/name** (QString): `completo`
**workspace-b1f1e949-4525-49d0-b0cd-2387aaaf78b1/noteSubFolderDockWidgetVisible** (bool): `true`
**workspace-b1f1e949-4525-49d0-b0cd-2387aaaf78b1/windowState** (QByteArray): `<binary data>`
**workspace-f8f2e239-da46-4cd3-9f91-a0920233714a/name** (QString): `mínimo`
**workspace-f8f2e239-da46-4cd3-9f91-a0920233714a/noteSubFolderDockWidgetVisible** (bool): `true`
**workspace-f8f2e239-da46-4cd3-9f91-a0920233714a/windowState** (QByteArray): `<binary data>`
**workspace-initial/noteSubFolderDockWidgetVisible** (bool): `true`
**workspace-initial/windowState** (QByteArray): `<binary data>`
**workspaces** (QStringList): `b1f1e949-4525-49d0-b0cd-2387aaaf78b1, f8f2e239-da46-4cd3-9f91-a0920233714a`

## System environment

**USER**: `gabrielriverabarreto`
**__CFBundleIdentifier**: `com.PBE.QOwnNotes`
**COMMAND_MODE**: `unix2003`
**LOGNAME**: `gabrielriverabarreto`
**PATH**: `/usr/bin:/bin:/usr/sbin:/sbin`
**SSH_AUTH_SOCK**: `/private/tmp/com.apple.launchd.AX1pAYSmCz/Listeners`
**SHELL**: `/bin/zsh`
**HOME**: `/Users/gabrielriverabarreto`
**__CF_USER_TEXT_ENCODING**: `0x1F5:0x0:0x8`
**TMPDIR**: `/var/folders/4j/7f13wwmn79gdbwfgvm7_sbpm0000gn/T/`
**XPC_SERVICE_NAME**: `application.com.PBE.QOwnNotes.53695866.53695869`
**XPC_FLAGS**: `0x0`
**QT_LOGGING_RULES**: `qt.network.ssl.warning=false`
</details>

**Comment by pbek** [comment]
Thank you for reporting.
Unfortunately I don't have macOS 11 (only 10) to reproduce this.
Do you maybe have a screenshot?

**Comment by pbek** [comment]
I've never seen that or heard from that before. What happens if you change the editor fonts in QOwnNotes?

**Comment by gabrivera** [comment]
It works! I changed it to Helvetica and the spaces appeared. Changed it to Arial, spaces appeared. Changed it back to the default ".AppleSystemUIFont (13)", spaces disappeared.

So it's probably the way the ".AppleSystemUIFont (13)" interacts with the app.

That interaction would be interesting to understand and fix 'at the root', but the quick-fix is so convenient that warranting a broader change seems overkill? Unless you'd like to address this further, I think your suggestion closed the issue!

Thanks!
